activiti-cloud-service-common/activiti-cloud-service-messaging-config/src/main/java/org/activiti/cloud/common/messaging/config/FunctionBindingConfiguration.java

```diff
@@ -188,4 +198,24 @@ public Object postProcessAfterInitialization(Object bean, String beanName) throw
     }
 
     public interface BindingResolver extends Function<String, String> {}
+
+    @Bean
+    @ConditionalOnClass(JacksonMapper.class)
+    @Primary
+    public JacksonMapper jacksonMapper(@Autowired(required = false) ObjectMapper objectMapper) {
+        //temporary workaround for https://github.com/spring-cloud/spring-cloud-function/issues/1159
+        ObjectMapper copiedMapper;
+        try {
+            copiedMapper = objectMapper.copy();
+            //logic from AlfrescoWebAutoConfiguration.configureObjectMapperForBigDecimal
+            copiedMapper.configOverride(BigDecimal.class).setFormat(JsonFormat.Value.forShape(JsonFormat.Shape.STRING));
+        } catch (Exception e) {
+            copiedMapper = new ObjectMapper();
+            copiedMapper.registerModule(new JavaTimeModule());
+            copiedMapper.configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false);
+            copiedMapper.configure(DeserializationFeature.FAIL_ON_TRAILING_TOKENS, true);
+            copiedMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
```

Comment:
Is it expected that this is called only in the case of exception? Shouldn't we always go through it?

Reply:
Ok, moved those settings outside of the catch. There is no risk since we modify copied object not a common one from the context